src/scanners/machine_uuid_scanner.go

```diff
@@ -36,24 +53,24 @@ func readSystemUUID() *strfmt.UUID {
 	return &ret
 }
 
-func readMotherboardSerial() *strfmt.UUID {
-	basedboard, err := ghw.Baseboard()
+func readMotherboardSerial(d Dependencies) *strfmt.UUID {
+	basedboard, err := d.Baseboard()
 	if err != nil {
 		log.Warnf("Could not find motherboard serial number: %s", err.Error())
 		return nil
 	}
 	value := basedboard.SerialNumber
-	if value == "" || value == ghw.UNKNOWN {
+	if value == "" || value == ghw.UNKNOWN || value == VmwareDefualtSerial {
 		log.Warn("Could not find motherboard serial number")
 		return nil
 	}
 	return md5GenerateUUID(value)
 }
 
-func ReadId() *strfmt.UUID {
-	ret := readMotherboardSerial()
+func ReadId(d Dependencies) *strfmt.UUID {
```

Comment:
Seems like all functions in this file require the Dependencies implementation.
Maybe this should turn into a class that is initialized with the dependencies?

Reply:
Besides ghw functions, there is no need for other mocks - since I don't think we need to mock md5.  It is a function that should always provides the same output given the same input.  There are not other functions in this file.

Reply:
Agree, assuming @ronniel1 meant struct :wink:

Reply:
class == stuct in this case
I did not talk about other mocks

Reply:
OK